src/components/SlateEditor/plugins/codeBlock/CodeBlock.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import styled from '@emotion/styled';
+import { Block, Document, Inline, Node } from 'slate';
+
+import Button from '@ndla/button';
+import { Cross } from '@ndla/icons/action';
+import { injectT } from '@ndla/i18n';
+import { Codeblock, getTitleFromFormat } from '@ndla/code';
+
+import { getSchemaEmbed } from '../../editorSchema';
+import { CodeBlockType, CodeBlockProps } from '../../../../interfaces';
+import EditCodeBlock from './EditCodeBlock';
+
+type ParentNode = Document | Block | Inline;
+
+const CodeDiv = styled.div`
+  cursor: pointer;
+`;
+
+interface RemoveCodeBlockProps {
+  handleRemove: () => void;
+}
+
+const RemoveCodeBlock: React.FC<RemoveCodeBlockProps> = ({ handleRemove }) => {
+  return (
+    <Button stripped onClick={handleRemove}>
+      <Cross />
+    </Button>
+  );
+}; // TODO:   Type 'Element' provides no match for the signature ...
+
+const getInfoFromNode = (node: Node) => {
+  const data = (node as ParentNode)?.data?.toJS() || {};
+  const codeBlock = data['code-block'] || node.text;
+
+  const code = codeBlock.code || data['code-content'] || '';
+  const format = codeBlock.format || data['code-format'] || 'text';
+
+  return {
+    model: {
+      code,
+      title: codeBlock.title || getTitleFromFormat(format),
+      format,
+    },
+    isFirstEdit: data['code-block'] === undefined,
+  };
+};
+
+const CodeBlock: React.FC<CodeBlockProps> = ({ attributes, editor, node }) => {
+  const { isFirstEdit, model } = getInfoFromNode(node);
+  const [editMode, setEditMode] = useState<boolean>(!model.code);
+  const [firstEdit, setFirstEdit] = useState<boolean>(isFirstEdit);
+
+  const toggleEditMode = () => {
+    setEditMode(!editMode);
+  };
+
+  const handleSave = (codeBlock: CodeBlockType) => {
+    const properties = {
+      data: { ...getSchemaEmbed(node), 'code-block': codeBlock },
+    };
+    setEditMode(false);
+    setFirstEdit(false);
+    editor.setNodeByKey(node.key, properties);
+  };
+
+  const handleRemove = () => {
+    editor.removeNodeByKey(node.key);
+    editor.focus();
+  };
+
+  const handleUndo = () => {
+    editor.unwrapBlockByKey(node.key, 'code-block');
+    editor.focus();
+  };
+
+  const onExit = () => {
+    setEditMode(false);
+    if (firstEdit) {
+      handleUndo();
+    }
+  };
+
+  return (
+    <CodeDiv
+      className="c-figure"
+      draggable={!editMode}
+      onClick={toggleEditMode}
+      role="button"
+      {...attributes}>
+      <Codeblock
+        actionButton={<RemoveCodeBlock handleRemove={handleRemove} />} // TODO:  actionButton?: FC | null;
```

Comment:
Her er vel egentlig problemet at typen til actionButton ikkje skulle vore en FC men heller en JSX.Element eller nokke sånt. For det er jo et element som skal sendes inn og ikkje en funksjon.

Reply:
Lager en PR i frontend-packages.

Eg får forresten stadig vekk "JavaScript heap out of memory" på denne branchen. Er det noko rart med koden som gjør dette?

Reply:
Fikk det selv når jeg linka frontend-packages med ed for å teste typer. Vet ikke hva som kunne forårsake det.

Reply:
Fikk "JavaScript heap out of memory" tidligere, men nå når jeg oppdaterte branchen (commit kommer straks så klarte jeg ikke recreate feilmeldingen :S vet heller ikke hva det kom av.